src/Lecture2.hs

```diff
@@ -70,7 +78,11 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: Int -> [Int] -> (Maybe Int, [Int])
+removeAt n list
+  | n < 0           = (Nothing, list)
+  | n > length list = (Nothing, list)
+  | otherwise       = ( Just(list !! n), take (n-1) list ++ drop (n+1) list)
```

Comment:
That's a correct implementation 👍🏻 
Alternatively, you can use a standard function [splitAt](https://hackage.haskell.org/package/base-4.16.0.0/docs/Prelude.html#v:splitAt) instead of writing manual recursion 🔃

It's not the most performant though as it traverses the list prefix 3 times while you traverse it only once